controlplane/kubeadm/internal/workload_cluster_rbac.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	rbac "k8s.io/api/rbac/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// NodeBootstrapTokenAuthGroup specifies which group a Node Bootstrap Token should be authenticated in
+	NodeBootstrapTokenAuthGroup = "system:bootstrappers:kubeadm:default-node-token"
+
+	// GetNodesClusterRoleName defines the name of the ClusterRole and ClusterRoleBinding to get nodes
+	GetNodesClusterRoleName = "kubeadm:get-nodes"
+
+	// NodesGroup defines the well-known group for all nodes.
+	NodesGroup = "system:nodes"
+
+	// KubeletConfigMapRolePrefix defines base kubelet configuration ConfigMap role prefix.
+	KubeletConfigMapRolePrefix = "kubeadm:"
+
+	// KubeletConfigMapName defines base kubelet configuration ConfigMap name.
+	KubeletConfigMapName = "kubelet-config-%d.%d"
+)
+
+// EnsureResource creates a resoutce if the target resource doesn't exist. If the resource exists already, this function will ignore the resource instead.
+func (w *Workload) EnsureResource(ctx context.Context, obj runtime.Object, testObj runtime.Object) error {
+	key, err := ctrlclient.ObjectKeyFromObject(obj)
+	if err != nil {
+		return errors.Wrap(err, "unable to derive key for resource")
+	}
+	err = w.Client.Get(ctx, key, testObj)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to determine if resource %s/%s already exists", key.Namespace, key.Name)
+	} else if err == nil {
+		// If object already exists, nothing left to do
+		return nil
+	}
+	if err := w.Client.Create(ctx, obj); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
```

Comment:
Is this just to be extra safe?

Reply:
Yup, though granted the chance of it happening is slim to nil.